src/handlers/shared/start.ts

```diff
@@ -42,73 +33,69 @@ export async function start(context: Context, issue: Context["payload"]["issue"]
-
-  // check for max and enforce max
-
-  if (assignedIssues.length - openedPullRequests.length >= maxConcurrentTasks) {
-    const log = logger.error("Too many assigned issues, you have reached your max limit", {
-      assignedIssues: assignedIssues.length,
-      openedPullRequests: openedPullRequests.length,
-      maxConcurrentTasks,
-    });
-    await addCommentToIssue(context, log?.logMessage.diff as string);
-    throw new Error(`Too many assigned issues, you have reached your max limit of ${maxConcurrentTasks} issues.`);
-  }
-
   // is it assignable?
 
   if (issue.state === ISSUE_TYPE.CLOSED) {
-    throw logger.error("This issue is closed, please choose another.", { issueNumber: issue.number });
+    throw new Error(logger.error("This issue is closed, please choose another.", { issueNumber: issue.number }).logMessage.raw);
   }
 
   const assignees = issue?.assignees ?? [];
 
   // find out if the issue is already assigned
   if (assignees.length !== 0) {
     const isCurrentUserAssigned = !!assignees.find((assignee) => assignee?.login === sender.login);
-    throw logger.error(
-      isCurrentUserAssigned ? "You are already assigned to this task." : "This issue is already assigned. Please choose another unassigned task.",
-      { issueNumber: issue.number }
+    throw new Error(
+      logger.error(
+        isCurrentUserAssigned ? "You are already assigned to this task." : "This issue is already assigned. Please choose another unassigned task.",
+        { issueNumber: issue.number }
+      ).logMessage.raw
     );
   }
 
   teammates.push(sender.login);
 
+  const toAssign = [];
   // check max assigned issues
   for (const user of teammates) {
-    await handleTaskLimitChecks(user, context, maxConcurrentTasks, logger, sender.login);
+    if (await handleTaskLimitChecks(user, context, maxConcurrentTasks, logger, sender.login)) {
+      toAssign.push(user);
+    }
+  }
+
+  let error: string | null = null;
+
+  if (toAssign.length === 0 && teammates.length > 1) {
+    error = "All teammates have reached their max task limit. Please close out some tasks before assigning new ones.";
+  } else if (toAssign.length === 0) {
+    error = "You have reached your max task limit. Please close out some tasks before assigning new ones.";
+  }
+
+  if (error) {
+    throw new Error(logger.error(error, { issueNumber: issue.number }).logMessage.raw);
   }
 
   // get labels
   const labels = issue.labels;
   const priceLabel = labels.find((label: Label) => label.name.startsWith("Price: "));
 
   if (!priceLabel) {
-    throw logger.error("No price label is set to calculate the duration", { issueNumber: issue.number });
+    throw new Error(logger.error("No price label is set to calculate the duration", { issueNumber: issue.number }).logMessage.raw);
   }
 
   const duration: number = calculateDurations(labels).shift() ?? 0;
+  const toAssignIds = toAssign.map(async (u) => await fetchUserId(context, u));
```

Comment:
check if any ids are null because that means the user doesn't exist (maybe we should throw error if that happens?)

Reply:
I went with throwing if null